frontend/components/puzzle4.jsx

```diff
@@ -34,9 +35,23 @@ class Puzzle4 extends React.Component {
     });
   }
 
+  submitAnswer() {
+    let answer = this.state.value;
+    let gameState = this.props.gameState;
+
+    // base case
+    if (Object.keys(gameState).length === 0) {
+      // TODO: can you explain why "if (gameState === {})" doesn't work?
```

Comment:
probably because they're not the _same_ object. triple equals checks that the two operands are the same reference for objects, as in, even if `gameState` is the empty object, they are different empty objects. i don't think `x === {}` will ever return true for any value of x.